update dockerfile to handle *.gemspec on build

lpmi-13 opened the pull request:
Because of the addition of the *.gemspec file into the build
process, it needs to be included in the Dockerfile to correctly
install all the gems.

Also, we're now suppressing a harmless warning about the git
discovery across file systems not being set.

desilinguist commented:
It works now! Thanks! Two minor observations:

1. I saw a couple of `fatal: not a git repository (or any of the parent directories` messages when I run `docker build` (see [build log](https://github.com/sigedu-org/website/files/3531193/build.log))



2. Is there a reason that it's installing and building some Ruby gems (e.g., nokogiri) _again_ for the `docker run` command (see [run log](https://github.com/sigedu-org/website/files/3531189/run.log))

lpmi-13 commented:
Having spent the last hour or so investigation what a `.gemspec` file even is (I'm really not a ruby developer), it seems obvious that we don't actually need the offending line in the `minimal-mistakes-jekyll.gemspec`, which was also the source of the weird git warnings (cause there's not git directory in the docker container).

I've updated this pull request to still include the .gemspec file but not need the lines causing the errors.

I also updated the README while I'm here, since the `-it` part of the docker commands don't effectively do anything.

lpmi-13 commented:
It looks like that `spec.files` part is only useful when actually building your own gems, so it shouldn't affect any local gem fetching and running.

The first docker command was (I assume) an old version of what used to run with that default jekyll/jekyll container, and so I've updated it, along with adding the `jekyll serve` like you mentioned.

In terms of the double gem fetching, I'm not seeing that locally, but I'm also deleting all the old docker images for a clean build. You could try running both `docker rm $(docker ps -aq)` and then `docker rmi $(docker images -q)` and then building and running again just to make sure you're using fresh images.

If you're still seeing double gem downloading, post the complete terminal command/output and I'll see if it's something I can reproduce (the whole point of docker is it should be deterministic if the commands are the same, so not really sure what could be causing that besides old images, but we'll give it a go anyway).

I've pushed those updates to this branch.

desilinguist commented:
I deleted all old images and re-ran the commands and still see double gem fetching. Here is the 
[build.log](https://github.com/sigedu-org/website/files/3556994/build.log) and here is the 
[run.log](https://github.com/sigedu-org/website/files/3556995/run.log).

lpmi-13 commented:
tried and failed to reproduce. I'm really at a loss as to what could be causing this...are you able to replicate the double fetching on a different machine? Also, what version of docker are you using? I'm seeing no issues with 19.03.0

I've put a call out to see if any docker or ruby specialists I know have encountered something like this before. I'm completely stumped.

desilinguist commented:
I am using `Docker version 19.03.1, build 74b1e89`. Oh well, perhaps it's just something with my setup if you can't replicate it. So, let's just go with it for now. Thanks again for taking so much time to help with this! It's much appreciated!
